Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -84,11 +88,28 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId = do
     QRB.updateStatus bookingId SRB.COMPLETED
     DriverStats.updateIdleTime driverId
     DriverStats.incrementTotalRidesAndTotalDist (cast ride.driverId) (fromMaybe 0 ride.chargeableDistance)
+    now <- getCurrentTime
+    let rideDate = utctDay $ addUTCTime (intToNominalDiffTime 19800) now
+    mbDailyRideStats <- QDailyStats.findByIdAndDate ride.driverId rideDate
+    case mbDailyRideStats of
+      Just _ -> do
+        QDailyStats.incrementDailyRidesAndDailyDist ride.driverId (fromMaybe 0 ride.chargeableDistance) rideDate
+      Nothing -> do
+        let driverDailyStats = createDriverDailyStats ride.driverId 1 (fromMaybe 0 ride.chargeableDistance) rideDate
+        QDailyStats.create driverDailyStats
     if driverInfo.active
       then QDFS.updateStatus ride.driverId DDFS.ACTIVE
       else QDFS.updateStatus ride.driverId DDFS.IDLE
   DLoc.updateOnRide driverId False
   SRide.clearCache $ cast driverId
+  where
+    createDriverDailyStats drivrId rideCount rideDist date =
+      DailyStats.DriverDailyStats
+        { driverId = drivrId,
```

Comment:
typo `drivrId`

Reply:
Done intentionally as i was getting error **This binding for ‘driverId’ shadows the existing binding**

Reply:
i don't think it will work as driverId we are getting is of type Id Driver but the DailyStats.DriverDailyStats needs Id Person

Reply:
ride is also in scope, right? You may use ride.driverId

Reply:
or you can do `cast driverId` also.